Stack/Opc.Ua.Core/Types/Encoders/XmlDecoder.cs

```diff
@@ -1074,7 +1074,7 @@ public ExpandedNodeId ReadExpandedNodeId(string fieldName)
                 EndField(fieldName);
             }
 
-            if (m_namespaceMappings != null && m_namespaceMappings.Length > value.NamespaceIndex)
+            if (m_namespaceMappings != null && m_namespaceMappings.Length > value.NamespaceIndex && !value.IsNull)
```

Comment:
If you check for null here, then I would expect that you have to check for null 5 lines below as well.

Reply:
I added the missing line

Reply:
thats about right, but still it wouldn't make sense to write it if the Identifier were IsNull==true?

Reply:
Yes, it makes sense to check it there as well